src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1378,43 +1363,64 @@ private static bool HasIdentityConversionInternal(TypeSymbol type1, TypeSymbol t
             Debug.Assert((object)type1 != null);
             Debug.Assert((object)type2 != null);
 
-            // https://github.com/dotnet/roslyn/issues/27961: If two types differ only by nullability and
-            // one has IsNullable unset and the other doesn't, which do we choose in inference?
-            // See NullableReferenceTypesTests.TypeInference_04.
-            var compareKind = includeNullability ?
-                TypeCompareKind.AllIgnoreOptions | TypeCompareKind.CompareNullableModifiersForReferenceTypes | TypeCompareKind.UnknownNullableModifierMatchesAny :
-                TypeCompareKind.AllIgnoreOptions;
+            var compareKind = TypeCompareKind.AllIgnoreOptions |
```

Comment:
>compareKind [](start = 16, length = 11)

nit: for readability, consider using `if () { compareKind |= TypeCompareKind.CompareNullableModifiersForReferenceTypes; ... }`

Also, can we assert that `unknownNullableModifierMatchesAny` is only set when `includeNullability` is set?